tools/exe/palletjack2ipahost

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env ruby
+
+# Create host objects in a FreeIPA or Red Hat IdM instance.
+#
+# This script will loop over all hosts in the PalletJack database and check
+# if they are registered on the IPA server. The missing hosts will be added.
+# A random password will be generated by IPA and stored in the pallet.
+#
+# If the IPA server is configured as authorative DNS server, records will
+# be created for the new host.
+#
+# Data model assumptions:
+# - Each domain corresponds uniquely to one IPv4 network
+
+require 'palletjack/tool'
+require 'palletjack/ipa'
+require 'yaml'
+
+class PalletJack2Ipahost < PalletJack::Tool
+
+  def process
+    @ipa_pallets = {}
+    # Grab a list of all hosts in IPA
+    host_find = PalletJack::Ipa::Command.new("host_find")
+
+    if host_find.error? then
+      puts "ERROR: " + host_find.error_name + " " + host_find.error_message
+      exit 1
+    end
+
+    # Then convert the result to an array of FQDN strings
+    ipa_hosts = []
+    host_find.results.each do |host|
+      ipa_hosts << host['fqdn'][0]
+    end
+
+    jack.each(kind: 'system') do |system|
+      next if ipa_hosts.include? system['net.dns.fqdn'] # Skip existing hosts
```

Comment:
I think that would depend on how we are going to handle provisioning / unprovisioning in general. Right now there will still be a handful of more-or-less manual steps to setup a new system, mostly running scripts. I imagine that we would want to wrap the tools in some kind of provisioning script, where we could delete a host object and creating a new one prior to re-installation.

Maybe this whole approach is wrong, and we should focus on a more one-shot kind of tool, more like create_system.